src/graph/entities/qg_edge.c

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2018-2019 Redis Labs Ltd. and Contributors
+ *
+ * This file is available under the Redis Labs Source Available License Agreement
+ */
+
+#include "qg_edge.h"
+#include "qg_node.h"
+#include "../../util/arr.h"
+#include <assert.h>
+
+QGEdge* QGEdge_New(QGNode *src, QGNode *dest, const char *relationship, const char *alias) {
+    QGEdge *e = rm_malloc(sizeof(QGEdge));
+    e->reltypes = array_new(const char*, 1);
+    e->reltypeIDs = array_new(uint, 1);
+    e->minHops = 1;
+    e->maxHops = 1;
+    e->alias = NULL;
+
+    return e;
```

Comment:
Missing id setting.
e->id = ?

Reply:
The ID is constructed by the QueryGraph; we can pass it in as an argument to this function.